coalib/processes/Processing.py

```diff
@@ -460,48 +476,56 @@ def yield_ignore_ranges(file_dict):
 
     :param file_dict: The file dictionary.
     """
+    ignores = ['ignore ', 'noqa', 'noqa ']
     for filename, file in file_dict.items():
-        start = None
-        bears = []
-        stop_ignoring = False
+        start_range = None
+        create_range = partial(get_bear_range, filename, file)
         for line_number, line in enumerate(file, start=1):
             # Before lowering all lines ever read, first look for the biggest
             # common substring, case sensitive: I*gnor*e, start i*gnor*ing,
             # N*oqa*.
-            if 'gnor' in line or 'oqa' in line:
-                line = line.lower()
-                if 'start ignoring ' in line:
-                    start = line_number
-                    bears = get_ignore_scope(line, 'start ignoring ')
-                elif 'stop ignoring' in line:
-                    stop_ignoring = True
-                    if start:
-                        yield (bears,
-                               SourceRange.from_values(
-                                   filename,
-                                   start,
-                                   1,
-                                   line_number,
-                                   len(file[line_number-1])))
-
-                else:
-                    for ignore_stmt in ['ignore ', 'noqa ', 'noqa']:
-                        if ignore_stmt in line:
-                            end_line = min(line_number + 1, len(file))
-                            yield (get_ignore_scope(line, ignore_stmt),
-                                   SourceRange.from_values(
-                                       filename,
-                                       line_number, 1,
-                                       end_line, len(file[end_line-1])))
-                            break
-
-        if stop_ignoring is False and start is not None:
-            yield (bears,
-                   SourceRange.from_values(filename,
-                                           start,
-                                           1,
-                                           len(file),
-                                           len(file[-1])))
+            if 'gnor' not in line and 'oqa' not in line:
```

Comment:
with the ignores constant from above this shouldn't be hardcoded or somebody might forget that